lib/_http_outgoing.js

```diff
@@ -685,10 +685,10 @@ function onError(msg, err, callback) {
 }
 
 function emitErrorNt(msg, err, callback) {
-  callback(err);
   if (typeof msg.emit === 'function' && !msg._closed) {
     msg.emit('error', err);
   }
+  callback(err);
```

Comment:
This is different from `Writable` where `callback` is invoked before `'error'`.

Reply:
But I see that before [PR#31818](https://github.com/nodejs/node/pull/31818)  is introduced, error event is invoked before the callback is made. Ref: [PR#L709-L712](https://github.com/nodejs/node/pull/31818/files#diff-48d21edbddb6e855d1ee5716c49bcdc0d913c11ee8a24a98ea7dbc60cd253556L709-L712)

Reply:
Was that an intentional breaking change to align the various APIs behaviors?

Reply:
Yes and it was flagged as semver-major.

Reply:
After referring to the implementation of `Writable`, I found that the `callback` is indeed called first and then the `error event` is emit. In order to be consistent with `Writable`, I will keep the previous call sequence.